bundle-workflow/src/test_workflow/dependency_installer.py

```diff
@@ -0,0 +1,62 @@
+import os
+import shutil
+
+from s3_utility import S3ReadWrite
+
+
+class DependencyInstaller:
+    """
+    Provides a dependency installer for the test suites.
+    """
+
+    def __init__(self, build):
+        self.build_id = build.id
+        self.version = build.version
+        self.arch = build.architecture
+        self.s3_utility = S3ReadWrite()
+        self.s3_maven_location = (
+            f"/builds/{self.version}/{self.build_id}/{self.arch}/maven"
+        )
+        self.s3_build_location = (
+            f"/builds/{self.version}/{self.build_id}/{self.arch}/plugins"
+        )
+
+    def installMavenDependencies(self, dependency_list):
+        """
+        Downloads the maven dependencies from S3 and puts them on the maven local path
+        for each dependency in the dependency_list.
+
+        :param dependency_list: list of dependencies for which the maven artifacts need to be downloaded.
+        """
+        for dependency in dependency_list:
```

Comment:
Could you help me understand how the dependency list is expected here? Seems like it expects the paths instead of dependency names. I think we should accept dependency names and generate paths to pull down from S3.

Reply:
It is the dependency name that is expected here. [Example tests](https://github.com/opensearch-project/opensearch-build/pull/318/files#diff-05f802ba74b1ca82bd5f6ef6241b4a5bc5a71035c756ff71b53ae963690475b1R22)

Reply:
If that's the case, this path seems wrong ` s3_path = f"{self.s3_maven_location}/{dependency}"` It exists under org/opensearch...